app/process/watcher/transfer/watcher.go

```diff
@@ -279,8 +286,17 @@ func (ctw Watcher) createFungiblePayload(transactionID string, receiver string,
 			err)
 		return nil, err
 	}
-	if properAmount.Cmp(nativeAsset.MinAmount) < 0 {
-		return nil, errors.New(fmt.Sprintf("Transfer Amount [%s] is less than Minimum Amount [%s].", properAmount, nativeAsset.MinAmount))
+
+	tokenPriceInfo, exist := ctw.pricingService.GetTokenPriceInfo(asset.ChainId, nativeAsset.Asset)
+	if !exist {
+		errMsg := fmt.Sprintf("[%s] - Couldn't get price info in USD for asset [%s].", transactionID, nativeAsset.Asset)
+		ctw.logger.Errorf(errMsg)
+		return nil, errors.New(errMsg)
+	}
+
+	amountInUsd := decimalHelper.GetAmountInUsd(tokenPriceInfo.UsdPrice, properAmount, ctw.assetsService, nativeAsset)
+	if amountInUsd.Cmp(tokenPriceInfo.MinAmountInUsdWithFee) < 0 {
+		return nil, errors.New(fmt.Sprintf("[%s] - Transfer Amount [%s] is less than Minimum Amount [%s].", transactionID, amountInUsd, nativeAsset.MinFeeAmountInUsd))
```

Comment:
Wouldn't it be better to only compare amounts here rather than doing calculations every time?